### PR TITLE
📖  Fix link for component-config-tutorial

### PR DESCRIPTION
**Pull request by rikatz**
component-config-tutorial link on the book is broken :)

It's pointing to componentconfig-tutorial, missing the dash (-).

Also, some links / references are broken, I'm force pointing here to the correct location in GH repo but can investigate further how should this be done

### CODE REVIEW

**Comment by rikatz** [comment]
/hold
Found some other places to fix :)

File: docs/book/src/component-config-tutorial/api-changes.md [hunk 1]
```diff
@@ -3,7 +3,7 @@
 This tutorial will show you how to create a custom configuration file for your
 project by modifying a project generated with the `--component-config` flag
 passed to the `init` command. The full tutorial's source can be found 
-[here][tutorial-source]. Make sure you've gone through the [installation 
+[here](https://github.com/kubernetes-sigs/kubebuilder/tree/master/docs/book/src/component-config-tutorial/testdata/project). Make sure you've gone through the [installation
```
Comment (camilamacedo86) [inline docs/book/src/component-config-tutorial/api-changes.md:6]:
```suggestion
[here](). Make sure you've gone through the [installation 
```

Comment (camilamacedo86) [inline docs/book/src/component-config-tutorial/api-changes.md:6]:
```suggestion
[here][tutorial-source]. Make sure you've gone through the [installation 
```

**Comment by rikatz** [comment]
flake?

**Review by camilamacedo86 — APPROVED** [review]
Thank you 🥇

**Pull request merged**